### PR TITLE
Clean up NameResolution.fs

### PR DESCRIPTION
**Pull request by kerams**
Factoring out some common bits.

### CODE REVIEW

File: src/Compiler/Facilities/DiagnosticsLogger.fs [hunk 1]
```diff
@@ -132,21 +132,21 @@ let ErrorWithSuggestions ((n, message), m, id, suggestions) =
 let ErrorEnabledWithLanguageFeature ((n, message), m, enabledByLangFeature) =
     DiagnosticEnabledWithLanguageFeature (n, message, m, enabledByLangFeature)
 
-let inline protectAssemblyExploration dflt f =
+let inline protectAssemblyExploration dflt ([<InlineIfLambda>] f) =
```
Comment (kerams) [inline src/Compiler/Facilities/DiagnosticsLogger.fs:135]: Plus this cheeky addition.

Comment (forki) [reply src/Compiler/Facilities/DiagnosticsLogger.fs:135]: Should probably go into different pull request

File: src/Compiler/Checking/NameResolution.fs [hunk 2]
```diff
@@ -4346,6 +4332,54 @@ let rec private EntityRefContainsSomethingAccessible (ncenv: NameResolver) m ad
...
+                hashSet
+
+            items
+            |> List.filter (fun x ->
+                 let demangledName = x.DemangledModuleOrNamespaceName
+                 IsInterestingModuleName demangledName && notFakeContainerModule ilTyconNames demangledName
+                 && EntityRefContainsSomethingAccessible ncenv m ad  x
+                 && not (IsTyconUnseen ad ncenv.g ncenv.amap m x)))
+
+let GetAccessibleSubModules g (ncenv: NameResolver) (modref: ModuleOrNamespaceRef) m ad =
```
Comment (psfinaki) [inline src/Compiler/Checking/NameResolution.fs:4358]: For future reference - it's technically a "submodule" (one word), so no need to capitalize M.

Comment (psfinaki) [reply src/Compiler/Checking/NameResolution.fs:4358]: (I know it was like this before :D)

**Review by psfinaki — APPROVED** [review]
Thanks @kerams. Keep rocking :)

**Review by T-Gro — APPROVED** [review]

**Pull request merged**